README.md

```diff
@@ -86,3 +152,42 @@ More information on secrets can be found [here](https://docs.github.com/en/actio
 [store_creds]: https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository
 [basic_auth]: https://docs.cloud.service.gov.uk/deploying_services/route_services/#example-route-service-to-add-username-and-password-authentication
 
+
+## Development
+
+### Running tests
+
+Writing and running automated tests ensures software is correct, reduces risk that users will experience bugs, and allows developers to move much faster as they can make changes with confidence that any breaks will be caught be the test suite. Once you have set up unit tests:
+
+```bash
+python -u -m pytest tests
+```
+
+### Running the code formatter
+
+The [code formatter](https://black.readthedocs.io/en/stable/) ensures source code is formatted consistently. Running the code formatter makes changes automatically that then need to be committed.
+
+```bash
+black ./
+```
+
+### Running the linter
+
+The linter checks for basic logic errors and bugs. Linting reports rule violations that must be corrected manually.  
+
+```bash
+pylint <Dashboard name>
+```
+
+###  Add reminder message to run formatter and linter before pushing to GitHub
```

Comment:
Minor: Has anybody used these instructions?  I've not seen anybody use it while we've been pairing.

Reply:
We tried to pick out the bits from the dashboard docs that would be useful for someone external. Happy to take feedback on sections that you don't think would be!

On the minor, I haven't but I'm about to set it up...

Reply:
Didn't work - removed!